### PR TITLE
E2C: Change permissions for navigating to Cloud Migration

### PR DESCRIPTION
**Pull request by joshhunt**
Previously, Server Admin was required to access E2C, however Server Admin is not available on Grafana Cloud so we can't test it there.

This broadens permission access for E2C to Org Admins, to allow access on Grafana Cloud. Note that at the moment there are no real APIs for this (everything's mocked in the frontend) and it's still behind feature flag.

### CODE REVIEW

File: pkg/services/navtree/navtreeimpl/admin.go [hunk 1]
```diff
@@ -135,7 +136,7 @@ func (s *ServiceImpl) getAdminNode(c *contextmodel.ReqContext) (*navtree.NavLink
 		configNodes = append(configNodes, storage)
 	}
 
-	if s.features.IsEnabled(ctx, featuremgmt.FlagOnPremToCloudMigrations) && c.SignedInUser.IsGrafanaAdmin {
+	if s.features.IsEnabled(ctx, featuremgmt.FlagOnPremToCloudMigrations) && c.SignedInUser.HasRole(org.RoleAdmin) {
```
Comment (eledobleefe) [inline pkg/services/navtree/navtreeimpl/admin.go:139]: 🤔 I get you're changing this to give any admin of each org permissions, right? Not clear in the issue description ;)

Comment (joshhunt) [reply pkg/services/navtree/navtreeimpl/admin.go:139]: You're right - I forgot to update the description.

Note, this isn't to properly finish that task we added today, this is just a temp-ish change.

Comment (joshhunt) [reply pkg/services/navtree/navtreeimpl/admin.go:139]: (updated pr description)

Comment (eledobleefe) [reply pkg/services/navtree/navtreeimpl/admin.go:139]: I understand this is kind of the same as mocking the API. We will review these parts when we have a proper API, right?

**Review by eledobleefe — APPROVED** [review]

File: pkg/services/navtree/navtreeimpl/admin.go (hunk 1 above)
Comment (eledobleefe) [reply pkg/services/navtree/navtreeimpl/admin.go:139]: Added the missing and mandatory labels ;) so the tests can pass

Comment (eledobleefe) [reply pkg/services/navtree/navtreeimpl/admin.go:139]: Sorry, not sure why my last comment has been duplicated 🤷

Comment (joshhunt) [reply pkg/services/navtree/navtreeimpl/admin.go:139]: When the actual APIs that can 'cause damage' are implemented, the backend developers will implement them with proper access control checks. These checks here are only for showing the UI in the nav tree, and not actually granting access to anything sensitive.

**Review by ashharrison90 — APPROVED** [review]
lgtm! 🙌

**Pull request merged**